dotCMS/src/main/java/com/dotcms/security/ContentSecurityPolicyUtil.java

```diff
@@ -0,0 +1,192 @@
+
+    private static String NONCE_REQUEST_ATTRIBUTE = "NONCE_REQUEST_ATTRIBUTE";
+
+    static {
+        contentSecurityPolicyResolvers = map(
+                "{script-src nonce}", new ContentSecurityPolicyResolver(
+                        (header) -> calculateNonceHeaderToScript(header),
+                        (headerConfig, htmlCode) -> calculateNonceToScript(headerConfig, htmlCode)
+                ),
+                "{style-src nonce}", new ContentSecurityPolicyResolver(
+                        (header) -> calculateNonceHeaderToStyle(header),
+                        (headerConfig, htmlCode) -> calculateNonceToStyle(headerConfig, htmlCode)
+                )
+        );
+    }
+
+    private ContentSecurityPolicyUtil() {
+    }
+
+    /**
+     *
+     * Set the <code>Content-Security-Policy</code> header according to the
+     * <code>ContentSecurityPolicy.header</code> config property, for example if the config property
+     * has the follow value <code>default-src 'self'</code>, then the header is going to be set with the value
+     *
+     * The <code>ContentSecurityPolicy.header</code> config property has two special values:
+     * - {script-src nonce}: You can use it like <code>script-src {script-src nonce}</code>, and a random
+     * nonce is going to be calculated and set to each script block in <code>htmlCode</code>
+     *
+     * - {style-src nonce}: You can use it like <code>style-src {style-src nonce}</code>, and a random
+     * nonce is going to be calculated and set to each style block in <code>htmlCode</code>
+     *
+     * When any of the special values to the <code>ContentSecurityPolicy.header</code> config property
+     * is used then a new nonce value is calculated and set it to any script block or style block
+     * in <code>htmlCode</code>, so if you have the follow html code:
+     *
+     * <pre>
+     *     <script>
+     *         console.log('This is a test');
+     *     </script>
+     *
+     *     <h1>THIS IS A TEST</h1>
+     * </pre>
+     *
+     * This method is going to return:
+     *
+     * <pre>
+     *     <script nonce='[random nonce value]'">
+     *         console.log('This is a test');
+     *     </script>
+     *
+     *     <h1>THIS IS A TEST</h1>
+     * </pre>
+     *
+     * Where <code>random nonce value</code> is a random string encode with Base 64 algorithm.
+     *
+     * @param htmlCode
+     * @param response
+     * @return
+     */
+
+    private static String calculateNonce() {
+        final String randomAlphanumeric = RandomStringUtils.randomAlphanumeric(RANDOM_STRING_LENGTH);
+        return Base64.encode(randomAlphanumeric.getBytes());
+    }
+
+    public static boolean isConfig() {
+        final String contentSecurityPolicyHeader = getContentSecurityPolicyHeader();
+
+        return UtilMethods.isSet(contentSecurityPolicyHeader);
+    }
+
+    private static String getContentSecurityPolicyHeader() {
+        return Config.getStringProperty(
```

Comment:
it seems to be handle in every request, we have seen in other context this could be clock, so it would be better to use the lazy

Reply:
done https://github.com/dotCMS/core/pull/21337/commits/b556eaf241e8a017d59a0927d77c4c025839bb2b